src/RpcServer/RpcServer.Wallet.cs

```diff
@@ -1,187 +1,92 @@
-                case "getunclaimedgas":
-                    {
-                        return GetUnclaimedGas();
-                    }
-                case "importprivkey":
-                    {
-                        string privkey = _params[0].AsString();
-                        return ImportPrivKey(privkey);
-                    }
-                case "listaddress":
-                    {
-                        return ListAddress();
-                    }
-                case "sendfrom":
-                    {
-                        UInt160 assetId = UInt160.Parse(_params[0].AsString());
-                        UInt160 from = _params[1].AsString().ToScriptHash();
-                        UInt160 to = _params[2].AsString().ToScriptHash();
-                        string value = _params[3].AsString();
-                        return SendFrom(assetId, from, to, value);
-                    }
-                case "sendmany":
-                    {
-                        int to_start = 0;
-                        UInt160 from = null;
-                        if (_params[0] is JString)
-                        {
-                            from = _params[0].AsString().ToScriptHash();
-                            to_start = 1;
-                        }
-                        JArray to = (JArray)_params[to_start];
-                        return SendMany(from, to);
-                    }
-                case "sendtoaddress":
-                    {
-                        UInt160 assetId = UInt160.Parse(_params[0].AsString());
-                        UInt160 scriptHash = _params[1].AsString().ToScriptHash();
-                        string value = _params[2].AsString();
-                        return SendToAddress(assetId, scriptHash, value);
-                    }
-                default:
-                    return null;
-            }
-        }
-
-        public void PostProcess(HttpContext context, string method, JArray _params, JObject result)
-        {
-            switch (method)
-            {
-                case "invoke":
-                case "invokefunction":
-                case "invokescript":
-                    ProcessInvoke(result);
-                    break;
-            }
-        }
-
-        private void ProcessInvoke(JObject result)
-        {
-            if (Wallet != null)
-            {
-                Transaction tx = Wallet.MakeTransaction(result["script"].AsString().HexToBytes());
-                ContractParametersContext context = new ContractParametersContext(tx);
-                Wallet.Sign(context);
-                if (context.Completed)
-                    tx.Witnesses = context.GetWitnesses();
-                else
-                    tx = null;
-                result["tx"] = tx?.ToArray().ToHexString();
-            }
-        }
+        private Wallet wallet;
 
         private void CheckWallet()
         {
-            if (Wallet is null)
+            if (wallet is null)
```

Comment:
Maybe we can have another Attribute for `RequireWallet` ?

Reply:
We don't need it. Because if you need a wallet, you should at least be able to access the wallet object. Now that you have access, you can easily determine if the wallet object is empty.